compiler/rustc_hir/src/hir.rs

```diff
@@ -94,6 +94,9 @@ pub enum LifetimeName {
     /// User wrote nothing (e.g., the lifetime in `&u32`).
     Implicit,
 
+    /// User wrote nothing and should have.
+    ImplicitMissing,
```

Comment:
I feel like this would make more sense as an `Implicit { missing: bool }`

Reply:
Having `missing: bool` increases the storage size of `LifetimeName`.

Reply:
Really? `LifetimeName::Param` already has `ParamName`, so I wouldn't imagine that an `Implicit { missing: bool }` variant would be larger than that.

And if it does...does it matter?

Reply:
@cjgillot can you change this for a perf run?